cmake/scripts/run_test.sh

```diff
@@ -64,27 +67,65 @@ run(){
 }
 
 diff() {
-  rm -f failing_diff
+  rm -f failing_diff*
+  rm -f diff*
   touch diff
 
+  test_successful=false
+
   #
-  # run diff or numdiff (if available) to determine whether files are the
-  # same. if they are not, output the first few lines of the output of
-  # numdiff, followed by the results of regular diff since the latter is
-  # just more readable
+  # Pick up main comparison file and all variants. A valid variant name is
+  # of the form [...].output.[STRING]
   #
+  for file in "${COMPARISON_FILE}" "${COMPARISON_FILE}".*; do
```

Comment:
It only matters because you mentioned in the documentation that files are searched in alphabetical order. It doesn't matter which way we search, so let's not care more than necessary :-)